src/main/java/com/google/api/codegen/transformer/ruby/RubyGapicSurfaceTransformer.java

```diff
@@ -369,15 +373,21 @@ private ViewModel generateTopLevelIndexView(ApiModel model, GapicProductConfig p
     ImmutableList.Builder<ModuleView> moduleViews = ImmutableList.builder();
 
     for (String moduleName : namer.getApiModules()) {
+      boolean isVersion = moduleName.equals(namer.getModuleVersionName());
+
       if (moduleName.equals(namer.getModuleServiceName())) {
         moduleViews.add(
             metadataTransformer
                 .generateReadmeMetadataView(model, productConfig, packageNamer)
                 .moduleName(moduleName)
                 .build());
-      } else if (includeVersionModule || !moduleName.equals(namer.getModuleVersionName())) {
+      } else if (includeVersionModule || !isVersion) {
         moduleViews.add(SimpleModuleView.newBuilder().moduleName(moduleName).build());
       }
+
+      if (!includeVersionModule && isVersion) {
```

Comment:
I probably should have done this, but would you mind adding a comment around here or at the method level so we don't forget why we break here? Something to the effect of if `includedVersionModule` is `false` then the list of returned modules will not include the version or anything after it.